backend/app/comment_handler.go

```diff
@@ -0,0 +1,168 @@
+		return
+	}
+
+	comment, err := app.client.CreateComment(comment)
+	if err != nil {
+		log.Fatal(err)
+		c.JSON(http.StatusInternalServerError, ErrorResponse{Error: "failed to create comment"})
+		return
+	}
+
+	c.JSON(http.StatusCreated,
+		Response{Message: "comment is created",
+			Payload: []models.Comment{comment}})
+
+}
+
+// GetComment gets a comment on a specific bug by id
+func (app *App) GetComment(c *gin.Context) {
+
+	id := c.Param("id")
+
+	if id == "" {
+		log.Println("comment ID is required")
+		c.JSON(http.StatusNotFound, ErrorResponse{Error: "comment ID is required"})
+		return
+	}
+
+	comment, err := app.client.GetComment(id)
+	if err != nil {
+		log.Println("comment is not found:", err)
+		c.JSON(http.StatusNotFound,
+			ErrorResponse{Error: "comment is not found"})
+		return
+
+	}
+
+	c.JSON(http.StatusOK,
+		Response{Message: "comment is found",
+			Payload: []models.Comment{comment}})
+
+}
+
+// DeleteComment deletes a comment on a specific bug by id
+func (app *App) DeleteComment(c *gin.Context) {
+
+	id := c.Param("id")
+
+	if id == "" {
+		log.Println("comment ID is required")
+		c.JSON(http.StatusBadRequest, ErrorResponse{Error: "comment ID is required"})
+		return
+	}
+
+	if err := app.client.DeleteComment(id); err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			log.Println("comment is not found:", err)
+			c.JSON(http.StatusNotFound, ErrorResponse{Error: "comment is not found"})
+		} else {
+			log.Fatal(err)
+			c.JSON(http.StatusInternalServerError, ErrorResponse{Error: "failed to delete comment"})
+		}
+		return
+	}
+
+	c.JSON(http.StatusOK, Response{Message: "comment is deleted"})
+}
+
+// ListComments lists all the comments for a specific bug
+func (app *App) ListComments(c *gin.Context) {
+
+	bugID := c.Query("bug_id")
+
+	if bugID == "" {
+		log.Println("bug id is not found")
+		c.JSON(http.StatusBadRequest,
+			ErrorResponse{Error: "bug id is not found"})
+		return
+	}
+
+	comments := app.client.ListComments(bugID)
+
+	if len(comments) == 0 {
+		log.Println("no comments are found for this bug")
+		c.JSON(http.StatusNotFound,
+			ErrorResponse{Error: "no comments are found for this bug"})
+		return
+	}
+
+	c.JSON(http.StatusOK,
+		Response{Message: "comment are found for the bug",
+			Payload: comments})
+}
+
+// UpdateComment updates a comment on a specific bug by id
+func (app *App) UpdateComment(c *gin.Context) {
+
+	comment := models.Comment{}
+	id := c.Param("id")
+
+	if err := c.ShouldBindJSON(&comment); err != nil {
+		log.Println("failed to update the comment:", err)
+		c.JSON(http.StatusBadRequest, ErrorResponse{Error: "invalid comment data"})
+		return
+	}
+
+	if err := comment.Validate(); err != nil {
+		log.Println("bugID, ownerID, and summary of the comment should be validated", err)
+		c.JSON(http.StatusBadRequest, ErrorResponse{Error: "comment is not validated"})
+		return
+	}
+
+	comment, err := app.client.UpdateComment(comment, id)
+	if err != nil {
```

Comment:
it should be a check `if errors.Is(err, gorm.ErrRecordNotFound)` then `if err == nil`